SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -789,6 +789,647 @@ void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry
+  su2double Gamma, auxXCoord, auxYCoord, auxZCoord, InverseDesign = 0.0, DeltaX, Coord_i, Coord_j, jp1Coord, *Coord = nullptr, MeanFuntion,
+  *Face_Normal = nullptr, auxArea, auxPress, Mach, Beta, R_Plane, Pressure_Inf,
+  ModVelocity_Inf, Velocity_Inf[3], factor, *Xcoord = nullptr, *Ycoord = nullptr, *Zcoord = nullptr,
+  *Pressure = nullptr, *FaceArea = nullptr, *EquivArea = nullptr, *TargetArea = nullptr, *NearFieldWeight = nullptr,
+  *Weight = nullptr, jFunction, jp1Function;
+  unsigned long jVertex, iVertex, iPoint, nVertex_NearField = 0, auxPoint,
+  *IdPoint = nullptr, *IdDomain = nullptr, auxDomain;
+  unsigned short iPhiAngle;
+  ofstream NearFieldEA_file; ifstream TargetEA_file;
+
+  su2double XCoordBegin_OF = config->GetEA_IntLimit(0);
+  su2double XCoordEnd_OF = config->GetEA_IntLimit(1);
+
+  unsigned short nDim = geometry->GetnDim();
+  su2double AoA = -(config->GetAoA()*PI_NUMBER/180.0);
+  su2double EAScaleFactor = config->GetEA_ScaleFactor(); // The EA Obj. Func. should be ~ force based Obj. Func.
+
+  Mach  = config->GetMach();
+  Gamma = config->GetGamma();
+  Beta = sqrt(Mach*Mach-1.0);
+  R_Plane = fabs(config->GetEA_IntLimit(2));
+  Pressure_Inf = config->GetPressure_FreeStreamND();
+  Velocity_Inf[0] = config->GetVelocity_FreeStreamND()[0];
+  Velocity_Inf[1] = config->GetVelocity_FreeStreamND()[1];
+  Velocity_Inf[2] = config->GetVelocity_FreeStreamND()[2];
+  ModVelocity_Inf = 0;
+  for (iDim = 0; iDim < 3; iDim++)
+    ModVelocity_Inf += Velocity_Inf[iDim] * Velocity_Inf[iDim];
+
+  factor = 4.0*sqrt(2.0*Beta*R_Plane) / (Gamma*Pressure_Inf*Mach*Mach);
+
+  if (rank == MASTER_NODE) cout << "Writing Equivalent Area files." << endl ;
+
+#ifndef HAVE_MPI
+
+  /*--- Compute the total number of points on the near-field ---*/
+
+  nVertex_NearField = 0;
+  for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++)
+    if (config->GetMarker_All_KindBC(iMarker) == NEARFIELD_BOUNDARY)
+      for (iVertex = 0; iVertex < geometry->GetnVertex(iMarker); iVertex++) {
+        iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+        Face_Normal = geometry->vertex[iMarker][iVertex]->GetNormal();
+        Coord = geometry->nodes->GetCoord(iPoint);
+
+        /*--- Using Face_Normal(z), and Coord(z) we identify only a surface,
+         note that there are 2 NEARFIELD_BOUNDARY surfaces ---*/
+
+        if ((Face_Normal[nDim-1] > 0.0) && (Coord[nDim-1] < 0.0)) nVertex_NearField ++;
+      }
+
+
+  /*--- Create an array with all the coordinates, points, pressures, face area,
+   equivalent area, and nearfield weight ---*/
+
+  Xcoord = new su2double[nVertex_NearField];
+  Ycoord = new su2double[nVertex_NearField];
+  Zcoord = new su2double[nVertex_NearField];
+  AzimuthalAngle = new short[nVertex_NearField];
+  IdPoint = new unsigned long[nVertex_NearField];
+  IdDomain = new unsigned long[nVertex_NearField];
+  Pressure = new su2double[nVertex_NearField];
+  FaceArea = new su2double[nVertex_NearField];
```

Comment:
If you want to modernize this a little you can use std::vector.
Have a look at CFlowOutput::SetAnalyzeSurface if you are interested.

Reply:
I have converted array to std::vector.